src/components/MultipleFileInputField/MultipleFileInputField.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form, FormControl, IconButton,
+} from '@edx/paragon';
+import classNames from 'classnames';
+import * as FontAwesome from '@fortawesome/free-solid-svg-icons/faTimes';
+import { getSizeInBytes, formatBytes } from './utils';
+
+const MultipleFileInputField = ({
+  input,
+  label,
+  type,
+  description,
+  disabled,
+  required,
+  className,
+  id,
+  meta: { touched, error },
+  ...props
+}) => {
+  const hasError = !!(touched && error);
+  const [size, setSize] = useState('0');
+  const [filesSizeError, setFilesSizeError] = useState(null);
+  const { name: inputName, onChange: inputOnChange, value: inputValues } = input;
+
+  const handleInputChange = (e) => {
+    const file = e.target.files[0];
+    const sizeOfPreviousFiles = getSizeInBytes(size);
+    const totalSize = sizeOfPreviousFiles + file.size;
+    if (totalSize > 2097152) {
+      setFilesSizeError('file you are trying makes total files size above 2mb.');
+    } else {
+      const fileReader = new FileReader();
+      fileReader.onload = () => {
+        const { name: fileName, size: fileSize } = file;
+        const { result: fileContents } = fileReader;
+        inputOnChange([...inputValues, { name: fileName, size: fileSize, contents: fileContents }]);
```

Comment:
Just curious, what's `inputValues` for?

Reply:
input values are provided by redux-form whenever these changes we have to call redux-forms' provided on change with new input value so redux-form automatically updates those. thats why I am calling its on change by appending new value to existing values.